components/style/values/specified/box.rs

```diff
@@ -177,3 +181,83 @@ impl Parse for WillChange {
         Ok(WillChange::AnimateableFeatures(custom_idents.into_boxed_slice()))
     }
 }
+
+#[cfg(feature = "gecko")]
+bitflags! {
+    #[derive(MallocSizeOf, ToComputedValue)]
+    /// These constants match Gecko's `NS_STYLE_TOUCH_ACTION_*` constants.
+    pub struct TouchAction: u8 {
+        /// `none` variant
+        const TOUCH_ACTION_NONE = structs::NS_STYLE_TOUCH_ACTION_NONE as u8;
+        /// `auto` variant
+        const TOUCH_ACTION_AUTO = structs::NS_STYLE_TOUCH_ACTION_AUTO as u8;
+        /// `pan-x` variant
+        const TOUCH_ACTION_PAN_X = structs::NS_STYLE_TOUCH_ACTION_PAN_X as u8;
+        /// `pan-y` variant
+        const TOUCH_ACTION_PAN_Y = structs::NS_STYLE_TOUCH_ACTION_PAN_Y as u8;
+        /// `manipulation` variant
+        const TOUCH_ACTION_MANIPULATION = structs::NS_STYLE_TOUCH_ACTION_MANIPULATION as u8;
+    }
+}
+
+#[cfg(feature = "gecko")]
+impl TouchAction {
+    #[inline]
+    /// Get default `touch-action` as `auto`
+    pub fn auto() -> ComputedTouchAction {
+        TouchAction::TOUCH_ACTION_AUTO
+    }
+}
+
+#[cfg(feature = "gecko")]
+impl ToCss for TouchAction {
+    fn to_css<W>(&self, dest: &mut W) -> fmt::Result where W: fmt::Write {
+        match *self {
+            TouchAction::TOUCH_ACTION_NONE => dest.write_str("none"),
+            TouchAction::TOUCH_ACTION_AUTO => dest.write_str("auto"),
+            TouchAction::TOUCH_ACTION_MANIPULATION => dest.write_str("manipulation"),
+            _ if self.contains(TouchAction::TOUCH_ACTION_PAN_X | TouchAction::TOUCH_ACTION_PAN_Y) => {
+                dest.write_str("pan-x pan-y")
+            },
+            _ if self.contains(TouchAction::TOUCH_ACTION_PAN_X) => {
+                dest.write_str("pan-x")
+            },
+            _ if self.contains(TouchAction::TOUCH_ACTION_PAN_Y) => {
+                dest.write_str("pan-y")
+            },
+            _ => panic!("invalid touch-action value"),
+        }
+    }
+}
+
+#[cfg(feature = "gecko")]
+impl Parse for TouchAction {
+    fn parse<'i, 't>(
+        _context: &ParserContext,
+        input: &mut Parser<'i, 't>
+    ) -> Result<TouchAction, ParseError<'i>> {
+        // FIXME: remove clone() when lifetimes are non-lexical
+        try_match_ident_ignore_ascii_case! { input,
```

Comment:
maybe put input to next line to alignment?

Reply:
I think it's fine to put the `input` in the same line of `try_match_ident_ignore_ascii_case` macro because the other part is used for pattern matching.
Besides, you can try to search `try_match_ident_ignore_ascii_case!` inside `servo`; then you can see all the other usages of it also put `input` in the first line.